File: platform/windows/uniffi/AdGuard.FilterListManager/MarshalLogic/FilterListManagerConstants.cs
namespace AdGuard.FilterListManager.MarshalLogic
{
    /// <summary>
    /// Constants holder form the Rust
    /// </summary>
    public class FilterListManagerConstants
    {
        public FilterListManagerConstants(int userRulesId,
            int customGroupId,
            int specialGroupId,
            int smallestFilterId)
        {
            UserRulesId = userRulesId;
            CustomGroupId = customGroupId;
            SpecialGroupId = specialGroupId;
            SmallestFilterId = smallestFilterId;
        }

        /// <summary>
        /// Gets the user rules identifier.
        /// </summary>
        public int UserRulesId { get; }

        /// <summary>
        /// Gets the custom group identifier.
        /// </summary>
        public int CustomGroupId { get; }

        /// <summary>
        /// Gets the special group identifier.
        /// </summary>
        public int SpecialGroupId { get; }

        /// <summary>
        /// Gets the smallest filter identifier.
        /// </summary>
        public int SmallestFilterId { get; }
    }
}

File: crates/ffi/src/platforms/android/lib/src/main/java/com/adguard/flm/FilterListManagerConstants.kt
package com.adguard.flm

/**
 * Structure used for passing constants through FFI
 */
data class FilterListManagerConstants(
    /**
     * Filter ID for *User rules* filter
     */
    val userRulesId: Int,

    /**
     * Group ID for special *custom filters group*
     */
    val customGroupId: Int,

    /**
     * Group ID for *special service filters*
     */
    val specialGroupId: Int,

    /**
     * Smallest possible filter_id. You can safely occupy any filter with an id lower than this number.
     * The library is guaranteed to never create a filter with this id
     */
    val smallestFilterId: Int
)
